keras/backend/theano_backend.py

```diff
@@ -357,8 +357,11 @@ def __call__(self, inputs):
         return self.function(*inputs)
 
 
-def function(inputs, outputs, updates=[]):
-    return Function(inputs, outputs, updates=updates)
+def function(inputs, outputs, updates=[], debug_nan=False):
+    mode = None
+    if _DEBUG_MODE == 'detect_nan':  # use suggested Theano detect_nan implementation
+        mode = theano.compile.MonitorMode(post_func=theano.compile.monitormode.detect_nan).excluding('local_elemwise_fusion', 'inplace')
```

Comment:
We have a "NanGuardMode" (you can just use that string) that is done for this:
http://deeplearning.net/software/theano/tutorial/nan_tutorial.html
http://deeplearning.net/software/theano/library/compile/nanguardmode.html

I think it would be better to use that then what you propose. We are making it skip some False positive error from time to time:

https://github.com/Theano/Theano/pull/3768